control-plane/agents/src/common/errors.rs

```diff
@@ -918,6 +920,12 @@ impl From<SvcError> for ReplyError {
                 source,
                 extra,
             },
+            SvcError::DrainNotAllowedWhenHAisDisabled {} => ReplyError {
+                kind: ReplyErrorKind::Unavailable,
```

Comment:
Is unavailable the right error kind here? Or should it be something more like forbidden or preconditions failed or invalid?

Reply:
Ahh, failed precondition might be a better fit here. Forbidden is not present. I would go with failed precondition.

Reply:
Done. Btw @tiagolobocastro I did not disable republish as core-agent ha tests use republish without HA components.